File: packages/camera/camera_platform_interface/test/types/focus_mode_test.dart
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:camera_platform_interface/src/types/focus_mode.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('FocusMode should contain 2 options', () {
    final values = FocusMode.values;

    expect(values.length, 2);
  });

  test("FocusMode enum should have items in correct index", () {
    final values = FocusMode.values;

    expect(values[0], FocusMode.auto);
    expect(values[1], FocusMode.locked);
  });

  test("serializeFocusMode() should serialize correctly", () {
    expect(serializeFocusMode(FocusMode.auto), "auto");
    expect(serializeFocusMode(FocusMode.locked), "locked");
  });

  test("deserializeFocusMode() should deserialize correctly", () {
    expect(deserializeFocusMode('auto'), FocusMode.auto);
    expect(deserializeFocusMode('locked'), FocusMode.locked);
  });
}

File: packages/camera/camera/android/src/test/java/io/flutter/plugins/camera/types/FocusModeTest.java
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.camera.types;

import static org.junit.Assert.assertEquals;

import io.flutter.plugins.camera.features.autofocus.FocusMode;
import org.junit.Test;

public class FocusModeTest {

  @Test
  public void getValueForString_returns_correct_values() {
    assertEquals(
        "Returns FocusMode.auto for 'auto'", FocusMode.getValueForString("auto"), FocusMode.auto);
    assertEquals(
        "Returns FocusMode.locked for 'locked'",
        FocusMode.getValueForString("locked"),
        FocusMode.locked);
  }

  @Test
  public void getValueForString_returns_null_for_nonexistant_value() {
    assertEquals(
        "Returns null for 'nonexistant'", FocusMode.getValueForString("nonexistant"), null);
  }

  @Test
  public void toString_returns_correct_value() {
    assertEquals("Returns 'auto' for FocusMode.auto", FocusMode.auto.toString(), "auto");
    assertEquals("Returns 'locked' for FocusMode.locked", FocusMode.locked.toString(), "locked");
  }
}
